### PR TITLE
Fix crash in public mode introduced by #370

### PR DESCRIPTION
**Pull request by maxpoulin64**
Oopsie. Just ran into it by accident.

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -67,13 +67,13 @@ function Client(manager, name, config) {
...
 	var client = this;
 
-	if (!client.config.token) {
-		client.updateToken(function() {
-			client.manager.updateUser(client.name, {token: client.config.token});
-		});
-	}
-
 	if (config) {
+		if (config.token) {
```
Comment (astorije) [inline src/client.js]: I guess the purpose of this PR is to ensure that the `config` (or `client.config` depending on where it is used) is not `null` or `undefined` so that `config.something` / `client.config.something` does not crash the app.

However, why did you move from `!client.config.token` to, effectively, `client.config.token`?

Comment (maxpoulin64) [inline src/client.js]: Huh, yeah I think I made a mistake there. I need to switch it back to `!`. It is a problem with `config` being undefined indeed.

Comment (maxpoulin64) [inline src/client.js]: @astorije updated.

**Comment by astorije** [comment]
Follow-up question: in which case(s) can we end up with `client.config` being `undefined` or `null`? You said you ran into this, so I'm curious what led you there.

**Comment by maxpoulin64** [comment]
`config` is a user's `.json` file. In public mode, when a user connects, a temporary anonymous client is made and this one doesn't have a config. No config means no JSON, so no token no saving.
